internal/bminventory/inventory.go

```diff
@@ -1597,6 +1597,12 @@ func (b *bareMetalInventory) refreshClusterHosts(ctx context.Context, cluster *c
 		return common.NewApiError(http.StatusInternalServerError, err)
 	}
 
+	err = b.detectAndStoreCollidingIPsForCluster(cluster.ID, tx)
```

Comment:
Why is it part of inventory.go ?

Reply:
By the same logic, why is `setMajorityGroupForCluster` part of inventory.go?

Reply:
I agree with Ori, it should not be included here. It need to be part of the monitor.

Reply:
but because the order is that cluster collect the data and it will take affect in the host maybe it make sense to add it to https://github.com/openshift/assisted-service/blob/b8b1e4d624327361184ab39581321759e8eb9216/internal/bminventory/inventory.go#L1125

Reply:
And if this is is the order then it can be called here as well?

Reply:
To summarize it, because cluster is collecting the data and the host will be affected we need to
1. collect 
2. refresh host
3. refresh cluster 
If this is the case then we have 3 places that we need to run this function from
1. https://github.com/openshift/assisted-service/blob/b8b1e4d624327361184ab39581321759e8eb9216/internal/bminventory/inventory.go#L1125
2. https://github.com/openshift/assisted-service/blob/b8b1e4d624327361184ab39581321759e8eb9216/internal/bminventory/inventory.go#L1640
3. https://github.com/openshift/assisted-service/blob/b8b1e4d624327361184ab39581321759e8eb9216/internal/cluster/cluster.go#L524
@ori-amizur what do you think?

Reply:
I think monitor is mandatory.  
This validation will not change as a result of user action.  Therefore, monitoring should be enough.